lib/api/controllers/util/requestAssertions.js

```diff
@@ -80,6 +80,15 @@ module.exports = {
   /**
    * @param {Request} request
    */
+  assertIdStartsNotUnderscore: function requestAssertHasId (request) {
+    if (request.input.resource._id.indexOf('_') === 0) {
```

Comment:
I propose a more efficient way to perform this check:

`if (request.input.resource.id.charAt(0) === '_') {`